src/SmartComponents/SystemAdvisor/SystemAdvisor.js

```diff
@@ -64,6 +64,7 @@ const BaseSystemAdvisor = () => {
   const [inventoryReportFetchStatus, setInventoryReportFetchStatus] =
     useState('pending');
   const [rows, setRows] = useState([]);
+  const [rowCount, setRowCount] = useState(0);
```

Comment:
Does it really have to be in a State?

Reply:
hmm.  it makes sense. Do you think useRef might be good option?

Reply:
if I put a simple variable inside component, the variable gets reinitialized every rerender.

Reply:
You could use `useMemo` and make it depend on the `rows`, but generally I feel  just using a simple variable that doesn't require too much computing should be fine.

Reply:
I have changed to useRef.

Reply:
The reason why i'm a bit wary about using a state for this is that it needs to be maintained, so every time the rows change it has to change another state as well.

Reply:
yeap, I have pushed new changes  to implement with useRef. I believe that useRef does not require computation as you said :).